ClosestCity/ClosestCity.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @par Licence
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+#include "ClosestCity.hpp"
+
+namespace ClosestCity {
+
+void removeCites(std::vector<City> &vec, const std::string &cityName) {
+    auto city = std::find_if(vec.begin(), vec.end(),
+                             [cityName](const City &city_on_list) {
+                                 return (city_on_list.getName() == (cityName));
+                             });
+    std::size_t cityNamePopulation = city->getPopulation();
+    for (unsigned element_nr = 0; element_nr < vec.size(); element_nr++) {
+        if (vec[element_nr].getPopulation() < cityNamePopulation) {
+            vec.erase(vec.begin() + element_nr);
+            element_nr--;
+        }
+    }
+}
+
+std::string getTheClosestCityThatIsBiggerThanMe(std::vector<City> vec,
+                                                const std::string &cityName) {
+    if (vec.size() > 1) {
+        std::string bestResultSoFar = "";
+
+        auto specifiedCity = std::find_if(
+            vec.begin(), vec.end(), [cityName](const City &city_on_list) {
+                return (city_on_list.getName() == cityName);
+            });
+
+        auto distanceToTheClosestCity =
+            specifiedCity->getDistanceToTheFarthest();
+
+        for (auto city : vec) {
+            if (city.getPopulation() > specifiedCity->getPopulation()) {
+                if (city.distanceFrom(*specifiedCity) <
+                    distanceToTheClosestCity) {
+                    bestResultSoFar = city.getName();
+                    distanceToTheClosestCity =
+                        city.distanceFrom(*specifiedCity);
+                }
+            }
+        }
+        return bestResultSoFar;
+    } else {
+        return " ";
+    }
+}
+std::string getAllBiggestCitiesInOrder(std::vector<City> vec,
+                                       std::string cityName) {
+    std::string cities_list = cityName;
+    std::string next_city_in_line = "";
+    while (cityName != " ") {
+        removeCites(vec, cityName);
```

Comment:
Please think about removing this line.
Perhaps it is not needed for performance.... But for now, you can merge it.